askama_parser/src/node.rs

```diff
@@ -667,6 +674,26 @@ impl<'a> BlockDef<'a> {
     }
 }
 
+fn check_end_name<'a>(
+    before: &'a str,
+    after: &'a str,
+    name: &'a str,
+    end_name: &'a str,
+    kind: &str,
+) -> IResult<&'a str, &'a str, ParsingError<&'a str>> {
+    let message = if name.is_empty() && !end_name.is_empty() {
```

Comment:
I think we should move the simple case up front:

```
if name == end_name {
    return Ok((after, end_name));
}
```

Then we can simplify the subsequent `if` because we've already yeeted out the case where both `name` and `end_name` are empty.

Also, please ditch the trailing commas in single-line `format!()` calls.

Reply:
Looks better indeed, good catch!